lms/services/digest/_models.py

```diff
@@ -0,0 +1,76 @@
+from dataclasses import dataclass
+from functools import cached_property
+from typing import List
+
+
+@dataclass
+class HUser:
+    """A class representing H users."""
+
+    h_userid: str
+    """The H userid like 'acct:name@lms.example.com'."""
+
+    display_name: str
+    """A human readable display name."""
+
+    email: str
+    """Email for the user."""
+
+
+@dataclass
+class HCourse:
+    """A class representing courses as groups in H."""
+
+    title: str
+    """Title for the course."""
+
+    authority_provided_id: str
+    """The course's authority provided id."""
+
+    aka: List[str]
+    """All authority provided ids which map to the course from groups etc."""
+
+    instructors: List[str]
+    """A list of H userids for users who are instructors for the course."""
+
+    annotations: List[dict] = None
+    """A list of annotations in the course."""
+
+    @cached_property
+    def learner_annotations(self):
+        """Get the annotations, but filtered to users who are learners."""
+
+        return [
+            annotation
+            for annotation in self.annotations
+            if annotation["author"]["userid"] not in self.instructors
+        ]
+
+
+@dataclass
+class Digest:
+    """A digest of activity for a given user."""
+
+    audience_user: HUser
+    """The user who is the intended recipient of the information."""
+
+    courses: List[HCourse]
+    """Courses which the digest is summarising."""
+
+    def serialize(self) -> dict:
+        """Get a plain dict representation of this digest."""
+
+        digest = {"total_annotations": 0, "courses": []}
+
+        for course in self.courses:
+            # We only count learner activity at the moment, not teachers
+            if course.learner_annotations:
+                digest["total_annotations"] += len(course.learner_annotations)
+                digest["courses"].append(
+                    {
+                        "title": course.title,
+                        "num_annotations": len(course.learner_annotations),
+                    }
+                )
+
+        return digest
```

Comment:
I like and dislike having this here.

Having this be separate from the service means we can test these bits very easily without having to kick off (and therefore fixture) a whole pile of stuff to inspect the output.

The downside is it means what we are actually sending to the email service is a function call farther away. We could do this inline.